packages/@aws-cdk/aws-apigateway/lib/authorizers/authorizer-base.ts

```diff
@@ -0,0 +1,91 @@
+import { Construct, Lazy, Resource, ResourceProps } from '@aws-cdk/core';
+import { IAuthorizer } from '../authorizer';
+import { AuthorizationType, Method } from '../method';
+import { IRestApi } from '../restapi';
+
+const AUTHORIZER_BASE_SYMBOL = Symbol.for('@aws-cdk/apigateway.AuthorizerBase');
+
+/**
+ * Base class for all custom authorizers
+ */
+export abstract class AuthorizerBase extends Resource implements IAuthorizer {
```

Comment:
I'd call this `CustomAuthorizer` (usually `Base` classes are internal)

Reply:
Whoops. Hold over from some refactoring.

Reply:
After a bit of refreshing my memory, I remember why I had to use `AuthorizerBase`.

First, not all authorizers are 'custom' authorizers. See APIGateway's [documentation on this](https://docs.aws.amazon.com/apigateway/api-reference/resource/method/#authorizationType). IAM and Cognito User Pools are not considered custom authorizers. At this time, only Lambda-based authorizers are considered 'custom', but there is no reason it can't be something else in the future.
The same nuance extends into implementation - implementing the two lambda (custom) authorizers and the cognito user pool would need a `CfnAuthorizer` defined while an IAM type authorizer needs only IAM principals and policies.

Second, I do need a common base class across all the authorizers in order to be bind the right parameters to the `Method` and be backwards compatible - 

https://github.com/aws/aws-cdk/blob/8b9d8e72bf5941246f2b1e92c7f0e2b156c5e60c/packages/%40aws-cdk/aws-apigateway/lib/method.ts#L132-L143

Thirdly, I cannot merge `AuthorizerBase` into `Authorizer` class because it creates a circular dependency at these two points -

https://github.com/aws/aws-cdk/blob/8b9d8e72bf5941246f2b1e92c7f0e2b156c5e60c/packages/%40aws-cdk/aws-apigateway/lib/authorizers/lambda.ts#L71

https://github.com/aws/aws-cdk/blob/8b9d8e72bf5941246f2b1e92c7f0e2b156c5e60c/packages/%40aws-cdk/aws-apigateway/lib/authorizer.ts#L18-L25

This means I need to create a separate class that acts as the base class. I can call it something else, like `LambdaAuthorizer`. But it doesn't have to be a lambda-based authorizer, although I suspect given their movement of APIGWv2, it's unlikely we'll have another one coming here.

Thoughts?